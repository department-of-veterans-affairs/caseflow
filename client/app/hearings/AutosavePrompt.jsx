import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loadingSymbolHtml } from '../components/RenderFunctions.jsx';
import moment from 'moment';

// This may go away in favor of the timestamp from updated record
const now = () => {
  return moment().
    format('h:mm a').
    replace(/(p|a)m/, '$1.m.');
};

export class AutosavePrompt extends React.Component {
  render() {
    const color = this.props.spinnerColor || '#323a45';

    if (this.props.saving) {
      return <div className="saving">Saving...
        <div className="loadingSymbol">{loadingSymbolHtml('', '100%', color)}</div>
      </div>;
    }

    return <span className="saving">Last saved at {now()}</span>;

  }
}

const mapStateToProps = (state) => ({
  saving: state.saving
});

export default connect(
  mapStateToProps
)(AutosavePrompt);

AutosavePrompt.propTypes = {
  saving: PropTypes.bool
};
