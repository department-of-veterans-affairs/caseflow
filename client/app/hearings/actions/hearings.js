import * as Constants from '../constants/constants';

export const handleServerError = (err) => ({
  type: Constants.HANDLE_SERVER_ERROR,
  payload: {
    err
  }
});
