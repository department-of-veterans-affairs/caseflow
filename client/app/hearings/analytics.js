export const debounceMs = 1000;

export const CATEGORIES = {
  HEARINGS_DAYS_PAGE: 'Hearing Days',
  DAILY_DOCKET_PAGE: 'Daily Docket',
  HEARING_WORKSHEET_PAGE: 'Hearing Worksheet'
};

export const ACTIONS = {
  EDIT_ISSUE: 'edit-issue',
  TOGGLE_PRELIMINARY_IMPRESSION: 'toggle-preliminary-impression',
  OPEN_HEARINGS_TAB: 'open-hearings-tab',
  OPEN_PAST_HEARING_DOCKET: 'open-past-hearing-docket',
  OPEN_CURRENT_HEARING_DOCKET: 'open-current-hearing-docket',

  // daily docket
  GO_BACK_TO_HEARING_DAYS: 'go-back-to-hearing-days',
  DOCKET_HEARING_PREPPED: 'hearing-prepped',
  DISPOSITION_SELECTED: 'disposition-selected',
  AOD_SELECTED: 'aod-selected',
  TRANSCRIPT_REQUESTED: 'transcript-requested',
  EVIDENCE_WINDOW_WAIVED: 'evidence-window-waived',

  // hearing worksheet
  OPEN_PAST_ISSUES_TAB: 'open-past-issues-tab',
  CLICK_ON_SAVE_TO_PDF: 'click-on-save-to-pdf',
  SELECT_VETERAN_FROM_DROPDOWN: 'select-veteran-from-dropdown',
  CLICK_ON_REVIEW_CLAIMS_FOLDER: 'click-on-review-claims-folder'
};
