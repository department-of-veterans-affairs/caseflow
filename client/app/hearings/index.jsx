import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import perflogger from 'redux-perf-middleware';
import thunk from 'redux-thunk';

import ConfigUtil from '../util/ConfigUtil';
import DocketsContainer from './DocketsContainer';
import DailyDocketContainer from './DailyDocketContainer';
import HearingWorksheetContainer from './HearingWorksheetContainer';
import { hearingsReducers, mapDataToInitialState } from './reducers/index';
import ScrollToTop from './util/ScrollTop';

const configureStore = (data) => {

  const middleware = [];

  if (!ConfigUtil.test()) {
    middleware.push(thunk, perflogger);
  }

  // This is to be used with the Redux Devtools Chrome extension
  // https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd
  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  const initialData = mapDataToInitialState(data);
  const store = createStore(
    hearingsReducers,
    initialData,
    composeEnhancers(applyMiddleware(...middleware))
  );

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./reducers/index', () => {
      store.replaceReducer(hearingsReducers);
    });
  }

  return store;
};

const Hearings = ({ hearings }) => {

  return <Provider store={configureStore(hearings)}>
    <div>
      <BrowserRouter>
        <div>
         <ScrollToTop />
          <Route exact path="/hearings/dockets"
            component={() => (
              <DocketsContainer
                veteran_law_judge={hearings.veteran_law_judge} />
            )}
          />

          <Route exact path="/hearings/dockets/:date"
            component={(props) => (
              <DailyDocketContainer
                veteran_law_judge={hearings.veteran_law_judge}
                date={props.match.params.date} />
            )}
          />

        <Route exact path="/hearings/:hearingId/worksheet"
            component={(props) => (
              <HearingWorksheetContainer
                veteran_law_judge={hearings.veteran_law_judge}
                hearingId={props.match.params.hearingId} />
            )}
          />
        </div>

      </BrowserRouter>
    </div>
  </Provider>;
};

export default Hearings;
