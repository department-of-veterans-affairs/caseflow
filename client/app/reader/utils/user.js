// External Dependencies
import { some } from 'lodash';

/**
 * Helper Method to Determine if the User is editing text
 */
export const isUserEditingText = () => some(
  document.querySelectorAll('input,textarea'),
  (elem) => document.activeElement === elem
);
