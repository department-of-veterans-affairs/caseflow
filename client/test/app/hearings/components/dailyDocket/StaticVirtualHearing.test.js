import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { axe } from 'jest-axe';

import { amaHearing } from 'test/data/hearings';
import { amaHearingPast } from 'test/data/hearings';
import { anyUser, vsoUser } from 'test/data/user';

import COPY from '../../../../../COPY';
import { StaticVirtualHearing } from 'app/hearings/components/dailyDocket/DailyDocketRowInputs';

describe('StaticVirtualHearing', () => {
  const defaultProps = {
    user: anyUser,
    hearing: amaHearing
  };

  const pastHearingProps = {
    user: anyUser,
    hearing: amaHearingPast
  };

  it('renders correctly', () => {
    const { container } = render(<StaticVirtualHearing {...defaultProps} />);

    expect(container).toMatchSnapshot();
  })

  it('renders past hearing correctly', () => {
    const { container } = render(<StaticVirtualHearing {...pastHearingProps} />);

    expect(container).toMatchSnapshot();
  })

  it('passes a11y testing', async () => {
    const { container } = render(<StaticVirtualHearing {...defaultProps} />);

    const results = await axe(container);
    expect(results).toHaveNoViolations();
  })

  it('passes a11y testing for past hearing', async () => {
    const { container } = render(<StaticVirtualHearing {...pastHearingProps} />);

    const results = await axe(container);
    expect(results).toHaveNoViolations();
  })

  it('displays correct label for host user', () => {
    const component = render(
      <StaticVirtualHearing {...defaultProps} user={{ userId: amaHearing.judgeId }}/>
    );

    expect(component).toMatchSnapshot();
    expect(screen.getByText(COPY.VLJ_VIRTUAL_HEARING_LINK_LABEL_FULL)).toBeInTheDocument()
  })

  it('displays correct label for guest user', () => {
    const component = render(
      <StaticVirtualHearing {...defaultProps} user={vsoUser}/>
    );

    expect(component).toMatchSnapshot();
    expect(screen.getByText(COPY.REPRESENTATIVE_VIRTUAL_HEARING_LINK_LABEL)).toBeInTheDocument()
  })
})
