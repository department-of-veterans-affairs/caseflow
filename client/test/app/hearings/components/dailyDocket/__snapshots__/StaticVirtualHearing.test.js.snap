// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticVirtualHearing displays correct label for guest user 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          href="https://care.evn.va.gov/bva-app/?join=1&media=&escalate=1&conference=BVA0000009@care.evn.va.gov&pin=2684353125#&role=guest"
          target="_blank"
        >
          <strong>
            Virtual Hearing Link
          </strong>
          <span
            data-css-32srrr=""
          >
             
            <svg
              class=""
              height="20"
              version="1.1"
              viewBox="0 0 18 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                stroke="none"
                stroke-width="1"
              >
                <g
                  fill="#0071bc"
                  fill-rule="nonzero"
                >
                  <g
                    transform="translate(1.000000, 1.000000)"
                  >
                    <path
                      d="M12.2767813,8.17242825 L11.7057188,8.17242825 C11.6223125,8.17242825 11.553875,8.19969476 11.500625,8.25448292 C11.4469375,8.30895216 11.4202188,8.37876082 11.4202188,8.46378132 L11.4202188,11.377344 C11.4202188,11.7779863 11.2805313,12.1208747 11.001,12.4062005 C10.7214688,12.6914624 10.3853438,12.8339818 9.99275,12.8339818 L2.5695,12.8339818 C2.1769375,12.8339818 1.84084375,12.6914624 1.56125,12.4062005 C1.28171875,12.1209066 1.14196875,11.7780182 1.14196875,11.377344 L1.14196875,3.80200456 C1.14196875,3.40139408 1.2816875,3.05853759 1.56125,2.77324374 C1.84084375,2.487918 2.1769375,2.34533485 2.5695,2.34533485 L8.8505625,2.34533485 C8.93396875,2.34533485 9.00240625,2.31794077 9.055875,2.26337585 C9.10934375,2.20874715 9.1360625,2.13900228 9.1360625,2.05394989 L9.1360625,1.47114806 C9.1360625,1.38625513 9.10934375,1.31638269 9.055875,1.26184966 C9.00240625,1.20722096 8.93396875,1.17995444 8.8505625,1.17995444 L2.5695,1.17995444 C1.86175,1.17995444 1.2565,1.43641913 0.7538125,1.94931663 C0.2513125,2.46218223 0,3.07984055 0,3.80210023 L0,11.3774715 C0,12.0996993 0.2513125,12.7174533 0.75384375,13.2301595 C1.25653125,13.7430888 1.86178125,13.9996811 2.56953125,13.9996811 L9.99275,13.9996811 C10.7004688,13.9996811 11.3057813,13.7430888 11.8084063,13.2301595 C12.3110625,12.7174533 12.562375,12.0997312 12.562375,11.3774715 L12.562375,8.46400456 C12.562375,8.37888838 12.535625,8.30904784 12.4819375,8.25448292 C12.4284063,8.19969476 12.3599688,8.17242825 12.2767813,8.17242825 Z"
                      id="external-link-outline-box"
                    />
                    <path
                      d="M15.8009687,0.172974943 C15.6879375,0.0576264237 15.5539375,0 15.3994062,0 L10.8313437,0 C10.6767187,0 10.5428125,0.0576264237 10.429875,0.172974943 C10.3168438,0.288259681 10.26025,0.42484738 10.26025,0.58267426 C10.26025,0.740501139 10.3168438,0.877120729 10.429875,0.992469248 L12.0000938,2.59491116 L6.183,8.53138952 C6.1234375,8.59207745 6.09375,8.661918 6.09375,8.74068793 C6.09375,8.81974487 6.12353125,8.88955353 6.183,8.95024146 L7.200125,9.98809112 C7.25959375,10.048779 7.32803125,10.0790433 7.40534375,10.0790433 C7.48265625,10.0790433 7.551125,10.0488109 7.6105625,9.98809112 L13.4276875,4.05170843 L14.9980313,5.65415034 C15.1109687,5.76949886 15.2448438,5.82718907 15.3995,5.82718907 C15.5541562,5.82718907 15.6880312,5.76949886 15.8010937,5.65415034 C15.914125,5.53883371 15.9705937,5.4022779 15.9705937,5.24445103 L15.9705937,0.58267426 C15.9705625,0.424751708 15.9139063,0.288259681 15.8009687,0.172974943 Z"
                      id="external-link-outline-arrow"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </span>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        href="https://care.evn.va.gov/bva-app/?join=1&media=&escalate=1&conference=BVA0000009@care.evn.va.gov&pin=2684353125#&role=guest"
        target="_blank"
      >
        <strong>
          Virtual Hearing Link
        </strong>
        <span
          data-css-32srrr=""
        >
           
          <svg
            class=""
            height="20"
            version="1.1"
            viewBox="0 0 18 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              stroke="none"
              stroke-width="1"
            >
              <g
                fill="#0071bc"
                fill-rule="nonzero"
              >
                <g
                  transform="translate(1.000000, 1.000000)"
                >
                  <path
                    d="M12.2767813,8.17242825 L11.7057188,8.17242825 C11.6223125,8.17242825 11.553875,8.19969476 11.500625,8.25448292 C11.4469375,8.30895216 11.4202188,8.37876082 11.4202188,8.46378132 L11.4202188,11.377344 C11.4202188,11.7779863 11.2805313,12.1208747 11.001,12.4062005 C10.7214688,12.6914624 10.3853438,12.8339818 9.99275,12.8339818 L2.5695,12.8339818 C2.1769375,12.8339818 1.84084375,12.6914624 1.56125,12.4062005 C1.28171875,12.1209066 1.14196875,11.7780182 1.14196875,11.377344 L1.14196875,3.80200456 C1.14196875,3.40139408 1.2816875,3.05853759 1.56125,2.77324374 C1.84084375,2.487918 2.1769375,2.34533485 2.5695,2.34533485 L8.8505625,2.34533485 C8.93396875,2.34533485 9.00240625,2.31794077 9.055875,2.26337585 C9.10934375,2.20874715 9.1360625,2.13900228 9.1360625,2.05394989 L9.1360625,1.47114806 C9.1360625,1.38625513 9.10934375,1.31638269 9.055875,1.26184966 C9.00240625,1.20722096 8.93396875,1.17995444 8.8505625,1.17995444 L2.5695,1.17995444 C1.86175,1.17995444 1.2565,1.43641913 0.7538125,1.94931663 C0.2513125,2.46218223 0,3.07984055 0,3.80210023 L0,11.3774715 C0,12.0996993 0.2513125,12.7174533 0.75384375,13.2301595 C1.25653125,13.7430888 1.86178125,13.9996811 2.56953125,13.9996811 L9.99275,13.9996811 C10.7004688,13.9996811 11.3057813,13.7430888 11.8084063,13.2301595 C12.3110625,12.7174533 12.562375,12.0997312 12.562375,11.3774715 L12.562375,8.46400456 C12.562375,8.37888838 12.535625,8.30904784 12.4819375,8.25448292 C12.4284063,8.19969476 12.3599688,8.17242825 12.2767813,8.17242825 Z"
                    id="external-link-outline-box"
                  />
                  <path
                    d="M15.8009687,0.172974943 C15.6879375,0.0576264237 15.5539375,0 15.3994062,0 L10.8313437,0 C10.6767187,0 10.5428125,0.0576264237 10.429875,0.172974943 C10.3168438,0.288259681 10.26025,0.42484738 10.26025,0.58267426 C10.26025,0.740501139 10.3168438,0.877120729 10.429875,0.992469248 L12.0000938,2.59491116 L6.183,8.53138952 C6.1234375,8.59207745 6.09375,8.661918 6.09375,8.74068793 C6.09375,8.81974487 6.12353125,8.88955353 6.183,8.95024146 L7.200125,9.98809112 C7.25959375,10.048779 7.32803125,10.0790433 7.40534375,10.0790433 C7.48265625,10.0790433 7.551125,10.0488109 7.6105625,9.98809112 L13.4276875,4.05170843 L14.9980313,5.65415034 C15.1109687,5.76949886 15.2448438,5.82718907 15.3995,5.82718907 C15.5541562,5.82718907 15.6880312,5.76949886 15.8010937,5.65415034 C15.914125,5.53883371 15.9705937,5.4022779 15.9705937,5.24445103 L15.9705937,0.58267426 C15.9705625,0.424751708 15.9139063,0.288259681 15.8009687,0.172974943 Z"
                    id="external-link-outline-arrow"
                  />
                </g>
              </g>
            </g>
          </svg>
        </span>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`StaticVirtualHearing displays correct label for host user 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          href="https://care.evn.va.gov/bva-app/?join=1&media=&escalate=1&conference=BVA0000009@care.evn.va.gov&pin=8600030#&role=host"
          target="_blank"
        >
          <strong>
            VLJ Virtual Hearing Link
          </strong>
          <span
            data-css-32srrr=""
          >
             
            <svg
              class=""
              height="20"
              version="1.1"
              viewBox="0 0 18 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                stroke="none"
                stroke-width="1"
              >
                <g
                  fill="#0071bc"
                  fill-rule="nonzero"
                >
                  <g
                    transform="translate(1.000000, 1.000000)"
                  >
                    <path
                      d="M12.2767813,8.17242825 L11.7057188,8.17242825 C11.6223125,8.17242825 11.553875,8.19969476 11.500625,8.25448292 C11.4469375,8.30895216 11.4202188,8.37876082 11.4202188,8.46378132 L11.4202188,11.377344 C11.4202188,11.7779863 11.2805313,12.1208747 11.001,12.4062005 C10.7214688,12.6914624 10.3853438,12.8339818 9.99275,12.8339818 L2.5695,12.8339818 C2.1769375,12.8339818 1.84084375,12.6914624 1.56125,12.4062005 C1.28171875,12.1209066 1.14196875,11.7780182 1.14196875,11.377344 L1.14196875,3.80200456 C1.14196875,3.40139408 1.2816875,3.05853759 1.56125,2.77324374 C1.84084375,2.487918 2.1769375,2.34533485 2.5695,2.34533485 L8.8505625,2.34533485 C8.93396875,2.34533485 9.00240625,2.31794077 9.055875,2.26337585 C9.10934375,2.20874715 9.1360625,2.13900228 9.1360625,2.05394989 L9.1360625,1.47114806 C9.1360625,1.38625513 9.10934375,1.31638269 9.055875,1.26184966 C9.00240625,1.20722096 8.93396875,1.17995444 8.8505625,1.17995444 L2.5695,1.17995444 C1.86175,1.17995444 1.2565,1.43641913 0.7538125,1.94931663 C0.2513125,2.46218223 0,3.07984055 0,3.80210023 L0,11.3774715 C0,12.0996993 0.2513125,12.7174533 0.75384375,13.2301595 C1.25653125,13.7430888 1.86178125,13.9996811 2.56953125,13.9996811 L9.99275,13.9996811 C10.7004688,13.9996811 11.3057813,13.7430888 11.8084063,13.2301595 C12.3110625,12.7174533 12.562375,12.0997312 12.562375,11.3774715 L12.562375,8.46400456 C12.562375,8.37888838 12.535625,8.30904784 12.4819375,8.25448292 C12.4284063,8.19969476 12.3599688,8.17242825 12.2767813,8.17242825 Z"
                      id="external-link-outline-box"
                    />
                    <path
                      d="M15.8009687,0.172974943 C15.6879375,0.0576264237 15.5539375,0 15.3994062,0 L10.8313437,0 C10.6767187,0 10.5428125,0.0576264237 10.429875,0.172974943 C10.3168438,0.288259681 10.26025,0.42484738 10.26025,0.58267426 C10.26025,0.740501139 10.3168438,0.877120729 10.429875,0.992469248 L12.0000938,2.59491116 L6.183,8.53138952 C6.1234375,8.59207745 6.09375,8.661918 6.09375,8.74068793 C6.09375,8.81974487 6.12353125,8.88955353 6.183,8.95024146 L7.200125,9.98809112 C7.25959375,10.048779 7.32803125,10.0790433 7.40534375,10.0790433 C7.48265625,10.0790433 7.551125,10.0488109 7.6105625,9.98809112 L13.4276875,4.05170843 L14.9980313,5.65415034 C15.1109687,5.76949886 15.2448438,5.82718907 15.3995,5.82718907 C15.5541562,5.82718907 15.6880312,5.76949886 15.8010937,5.65415034 C15.914125,5.53883371 15.9705937,5.4022779 15.9705937,5.24445103 L15.9705937,0.58267426 C15.9705625,0.424751708 15.9139063,0.288259681 15.8009687,0.172974943 Z"
                      id="external-link-outline-arrow"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </span>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        href="https://care.evn.va.gov/bva-app/?join=1&media=&escalate=1&conference=BVA0000009@care.evn.va.gov&pin=8600030#&role=host"
        target="_blank"
      >
        <strong>
          VLJ Virtual Hearing Link
        </strong>
        <span
          data-css-32srrr=""
        >
           
          <svg
            class=""
            height="20"
            version="1.1"
            viewBox="0 0 18 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              stroke="none"
              stroke-width="1"
            >
              <g
                fill="#0071bc"
                fill-rule="nonzero"
              >
                <g
                  transform="translate(1.000000, 1.000000)"
                >
                  <path
                    d="M12.2767813,8.17242825 L11.7057188,8.17242825 C11.6223125,8.17242825 11.553875,8.19969476 11.500625,8.25448292 C11.4469375,8.30895216 11.4202188,8.37876082 11.4202188,8.46378132 L11.4202188,11.377344 C11.4202188,11.7779863 11.2805313,12.1208747 11.001,12.4062005 C10.7214688,12.6914624 10.3853438,12.8339818 9.99275,12.8339818 L2.5695,12.8339818 C2.1769375,12.8339818 1.84084375,12.6914624 1.56125,12.4062005 C1.28171875,12.1209066 1.14196875,11.7780182 1.14196875,11.377344 L1.14196875,3.80200456 C1.14196875,3.40139408 1.2816875,3.05853759 1.56125,2.77324374 C1.84084375,2.487918 2.1769375,2.34533485 2.5695,2.34533485 L8.8505625,2.34533485 C8.93396875,2.34533485 9.00240625,2.31794077 9.055875,2.26337585 C9.10934375,2.20874715 9.1360625,2.13900228 9.1360625,2.05394989 L9.1360625,1.47114806 C9.1360625,1.38625513 9.10934375,1.31638269 9.055875,1.26184966 C9.00240625,1.20722096 8.93396875,1.17995444 8.8505625,1.17995444 L2.5695,1.17995444 C1.86175,1.17995444 1.2565,1.43641913 0.7538125,1.94931663 C0.2513125,2.46218223 0,3.07984055 0,3.80210023 L0,11.3774715 C0,12.0996993 0.2513125,12.7174533 0.75384375,13.2301595 C1.25653125,13.7430888 1.86178125,13.9996811 2.56953125,13.9996811 L9.99275,13.9996811 C10.7004688,13.9996811 11.3057813,13.7430888 11.8084063,13.2301595 C12.3110625,12.7174533 12.562375,12.0997312 12.562375,11.3774715 L12.562375,8.46400456 C12.562375,8.37888838 12.535625,8.30904784 12.4819375,8.25448292 C12.4284063,8.19969476 12.3599688,8.17242825 12.2767813,8.17242825 Z"
                    id="external-link-outline-box"
                  />
                  <path
                    d="M15.8009687,0.172974943 C15.6879375,0.0576264237 15.5539375,0 15.3994062,0 L10.8313437,0 C10.6767187,0 10.5428125,0.0576264237 10.429875,0.172974943 C10.3168438,0.288259681 10.26025,0.42484738 10.26025,0.58267426 C10.26025,0.740501139 10.3168438,0.877120729 10.429875,0.992469248 L12.0000938,2.59491116 L6.183,8.53138952 C6.1234375,8.59207745 6.09375,8.661918 6.09375,8.74068793 C6.09375,8.81974487 6.12353125,8.88955353 6.183,8.95024146 L7.200125,9.98809112 C7.25959375,10.048779 7.32803125,10.0790433 7.40534375,10.0790433 C7.48265625,10.0790433 7.551125,10.0488109 7.6105625,9.98809112 L13.4276875,4.05170843 L14.9980313,5.65415034 C15.1109687,5.76949886 15.2448438,5.82718907 15.3995,5.82718907 C15.5541562,5.82718907 15.6880312,5.76949886 15.8010937,5.65415034 C15.914125,5.53883371 15.9705937,5.4022779 15.9705937,5.24445103 L15.9705937,0.58267426 C15.9705625,0.424751708 15.9139063,0.288259681 15.8009687,0.172974943 Z"
                    id="external-link-outline-arrow"
                  />
                </g>
              </g>
            </g>
          </svg>
        </span>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`StaticVirtualHearing renders correctly 1`] = `
<div>
  <div>
    <a
      href="https://care.evn.va.gov/bva-app/?join=1&media=&escalate=1&conference=BVA0000009@care.evn.va.gov&pin=2684353125#&role=guest"
      target="_blank"
    >
      <strong>
        Virtual Hearing Link
      </strong>
      <span
        data-css-32srrr=""
      >
         
        <svg
          class=""
          height="20"
          version="1.1"
          viewBox="0 0 18 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              fill="#0071bc"
              fill-rule="nonzero"
            >
              <g
                transform="translate(1.000000, 1.000000)"
              >
                <path
                  d="M12.2767813,8.17242825 L11.7057188,8.17242825 C11.6223125,8.17242825 11.553875,8.19969476 11.500625,8.25448292 C11.4469375,8.30895216 11.4202188,8.37876082 11.4202188,8.46378132 L11.4202188,11.377344 C11.4202188,11.7779863 11.2805313,12.1208747 11.001,12.4062005 C10.7214688,12.6914624 10.3853438,12.8339818 9.99275,12.8339818 L2.5695,12.8339818 C2.1769375,12.8339818 1.84084375,12.6914624 1.56125,12.4062005 C1.28171875,12.1209066 1.14196875,11.7780182 1.14196875,11.377344 L1.14196875,3.80200456 C1.14196875,3.40139408 1.2816875,3.05853759 1.56125,2.77324374 C1.84084375,2.487918 2.1769375,2.34533485 2.5695,2.34533485 L8.8505625,2.34533485 C8.93396875,2.34533485 9.00240625,2.31794077 9.055875,2.26337585 C9.10934375,2.20874715 9.1360625,2.13900228 9.1360625,2.05394989 L9.1360625,1.47114806 C9.1360625,1.38625513 9.10934375,1.31638269 9.055875,1.26184966 C9.00240625,1.20722096 8.93396875,1.17995444 8.8505625,1.17995444 L2.5695,1.17995444 C1.86175,1.17995444 1.2565,1.43641913 0.7538125,1.94931663 C0.2513125,2.46218223 0,3.07984055 0,3.80210023 L0,11.3774715 C0,12.0996993 0.2513125,12.7174533 0.75384375,13.2301595 C1.25653125,13.7430888 1.86178125,13.9996811 2.56953125,13.9996811 L9.99275,13.9996811 C10.7004688,13.9996811 11.3057813,13.7430888 11.8084063,13.2301595 C12.3110625,12.7174533 12.562375,12.0997312 12.562375,11.3774715 L12.562375,8.46400456 C12.562375,8.37888838 12.535625,8.30904784 12.4819375,8.25448292 C12.4284063,8.19969476 12.3599688,8.17242825 12.2767813,8.17242825 Z"
                  id="external-link-outline-box"
                />
                <path
                  d="M15.8009687,0.172974943 C15.6879375,0.0576264237 15.5539375,0 15.3994062,0 L10.8313437,0 C10.6767187,0 10.5428125,0.0576264237 10.429875,0.172974943 C10.3168438,0.288259681 10.26025,0.42484738 10.26025,0.58267426 C10.26025,0.740501139 10.3168438,0.877120729 10.429875,0.992469248 L12.0000938,2.59491116 L6.183,8.53138952 C6.1234375,8.59207745 6.09375,8.661918 6.09375,8.74068793 C6.09375,8.81974487 6.12353125,8.88955353 6.183,8.95024146 L7.200125,9.98809112 C7.25959375,10.048779 7.32803125,10.0790433 7.40534375,10.0790433 C7.48265625,10.0790433 7.551125,10.0488109 7.6105625,9.98809112 L13.4276875,4.05170843 L14.9980313,5.65415034 C15.1109687,5.76949886 15.2448438,5.82718907 15.3995,5.82718907 C15.5541562,5.82718907 15.6880312,5.76949886 15.8010937,5.65415034 C15.914125,5.53883371 15.9705937,5.4022779 15.9705937,5.24445103 L15.9705937,0.58267426 C15.9705625,0.424751708 15.9139063,0.288259681 15.8009687,0.172974943 Z"
                  id="external-link-outline-arrow"
                />
              </g>
            </g>
          </g>
        </svg>
      </span>
    </a>
  </div>
</div>
`;
